error.go

```diff
@@ -36,6 +53,86 @@ func (e internalError) InternalError() error {
 	return e.error
 }
 
+// Code set the error's translation code.
+func (e Error) Code(code string) Error {
```

Comment:
I think the Go naming convention for getters and setters is:
* getter: `Code() string`
* setter: `SetCode(code string)`

Reply:
Yea, I change it.